R/standalone-types-check.R

```diff
@@ -459,11 +459,16 @@ check_formula <- function(x,
 
 check_character <- function(x,
                             ...,
+                            allow_na = FALSE,
                             allow_null = FALSE,
                             arg = caller_arg(x),
                             call = caller_env()) {
+
   if (!missing(x)) {
-    if (is_character(x)) {
+    if (is_character(x) & allow_na) {
+      return(invisible(NULL))
+    }
+    if(is.character(x) & !(TRUE %in% is.na(x)) & !allow_na){
```

Comment:
These checks should use the scalar boolean operators, e.g. `&&`. (Also missing a space between `if` and `(`.)

Reply:
A simpler way of expressing `TRUE %in% is.na(x)` is `any(is.na(x))`.

Reply:
To solve the error message I think I'd structure this code like this:

```r
if (is_character(x)) {
  if (!allow_na && any(is.na(x))) {
    abort(...)
  }
  return(invisible(NULL))
}
```

With a specially crafted error message in the `abort()` call.